src/components/Croupier.tsx

```diff
@@ -175,7 +190,9 @@ const Croupier = () => {
                 gamePieceAppearance={uiState.cardPositions[card.uid]}
                 onClick={emit.playerCardClick(card)}
                 options={{ showAbilityButton: gameState.turn.player === player.uid && !hasTag("stagingEvent") }}
-                isHighlighted={hasTag("usingAbility") && test.playerCardClick(card)}
+                isHighlighted={
+                  (hasTag("usingAbility") && test.playerCardClick(card)) || hasStageHighlight(gameState, card.uid)
```

Comment:
what is we will have guards for that kind of staff?

```ts
guards.isOnStage(gameState, card)
```

Reply:
a step forward is to have an wrapper above guards to skip passing gameState:

```ts
guards.isOnStage(card)
```

Reply:
Yeah I struggled to think of a good location to put this, we can add a guard but it will not be enough. At the moment the stage cards are only highlighted when the event is a buy event, so then we need to check for two guards here, making it almost the same as just checking the state in Croupier itself. I will see what it looks like after trying out guards and revert back

Reply:
it's OK IMO. We could use tags:

```ts
hasTag('stage') || guards.isOnStage(card)
```


It's now about saving symbols, but about keeping react component having less logic.